framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -3194,6 +3194,43 @@ VkResult VulkanReplayConsumerBase::OverrideQueueSubmit(PFN_vkQueueSubmit func,
         GetDeviceTable(queue_info->handle)->QueueWaitIdle(queue_info->handle);
     }
 
+    // Check whether any of the submitted command lists buffers are frame boundaries.
+    if (screenshot_handler_ != nullptr)
+    {
+        bool is_frame_boundary = false;
+        for (uint32_t i = 0; i < submitCount; ++i)
```

Comment:
I saw a loop looking for this elsewhere outside this file. Can the several loops be refactored into a reusable function with a self-documenting name?

Reply:
My bad for cramming my review into late night tired time. Great that you have it covered.